src/extension/iSamples_resultList.js

```diff
@@ -3,76 +3,23 @@ import React from "react";
 import cx from "classnames";
 import Table from 'components/react_table';
 import CesiumMap from "components/cesium_map/cesium_UI";
+import ButGroup from 'components/ButGroup';
+import { store } from "redux/store";
 
-// Functional conponent to render button group
-const ButGroup = (props) => {
-  const { switchFormat, bootstrapCss, active } = props;
-
-  return (
-    <div className={cx({ "paginationBox": bootstrapCss })}>
-      <div className={cx({ "swithButtonBox": bootstrapCss })}>
-        <button className={cx({
-          "btn": bootstrapCss,
-          "btn-default": bootstrapCss,
-          "btn-sm": bootstrapCss,
-          "pull-left": bootstrapCss,
-          "margin-right-xs": bootstrapCss,
-          "active": active === 'List'
-        })} onClick={() => switchFormat("List")}>List</button>
-        <button className={cx({
-          "btn": bootstrapCss,
-          "btn-default": bootstrapCss,
-          "btn-sm": bootstrapCss,
-          "pull-left": bootstrapCss,
-          "margin-right-xs": bootstrapCss,
-          "active": active === 'Table'
-        })} onClick={() => switchFormat("Table")}>Table</button>
-        <button className={cx({
-          "btn": bootstrapCss,
-          "btn-default": bootstrapCss,
-          "btn-sm": bootstrapCss,
-          "pull-left": bootstrapCss,
-          "margin-right-xs": bootstrapCss,
-          "active": active === 'Map'
-        })} onClick={() => switchFormat("Map")}>Map</button>
-      </div>
-      {props.children}
-    </div>
-  );
-};
 
 class ResultList extends React.Component {
-  constructor() {
-    super();
-    this.state = { facet: "List" };
-  }
 
-  // set the initial camera position
-  componentDidMount() {
-    this.setState(this.props.view);
-  }
-
-  // React life cycle method to initialzie state
-  static getDerivedStateFromProps(props, state) {
-    return props.view;
-  }
 
+  // prevent unnecessary rerendering
   shouldComponentUpdate(nextProps) {
-    if (JSON.stringify(this.props) !== JSON.stringify(nextProps)) {
-      this.switchView(nextProps.view.facet);
-      this.setState(nextProps.view);
+    this.switchView(store.getState()['query']['view']['facet']);
+    if (JSON.stringify(nextProps) !== JSON.stringify(this.props)) {
       return true;
```

Comment:
We still need `shouldComponentUpdate` to indicate when the component should be rerenderred. Because this component still has dependent variables from props. Without this function, the `resultList` component would render multiple times even though the prop doesn't change. This is caused by react design. The parent component changes and all child components would rerender too.